src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Helpers/PolicyHelper.cs

```diff
@@ -41,5 +45,152 @@ public static List<string> GetRolesWithAccess(XacmlPolicy policy)
+        /// <returns>List of resource policies</returns>
+        public static List<ResourcePolicy> GetResourcePoliciesFromXacmlPolicy(XacmlPolicy policy)
+        {
+            Dictionary<string, ResourcePolicy> resourcePolicies = new Dictionary<string, ResourcePolicy>();
+
+            foreach (XacmlRule rule in policy.Rules)
+            {
+                if (rule.Effect.Equals(XacmlEffectType.Permit) && rule.Target != null)
+                {
+                    List<string> policyKeys = GetResourcePoliciesFromRule(resourcePolicies, rule);
+                    List<RoleGrant> roles = GetRolesFromRule(rule);
+                    List<ResourceAction> actions = GetActionsFromRule(rule, roles);
+
+                    foreach (string policyKey in policyKeys)
+                    {
+                        ResourcePolicy resourcePolicy = resourcePolicies.GetValueOrDefault(policyKey);
+
+                        if (policy.Description != null && resourcePolicy.Description == null)
+                        {
+                            resourcePolicy.Description = new LocalizedText(rule.Description, rule.Description, rule.Description);
+                        }
+
+                        if (resourcePolicy.Actions == null)
+                        {
+                            resourcePolicy.Actions = new List<ResourceAction>();
+                            resourcePolicy.Actions.AddRange(actions);
+                        }
+                        else
+                        {
+                            foreach (ResourceAction resourceAction in actions)
+                            {
+                                if (!resourcePolicy.Actions.Any(action => action.Match.Value == resourceAction.Match.Value && action.Match.Id == resourceAction.Match.Id))
+                                {
+                                    resourcePolicy.Actions.Add(resourceAction);
+                                }
+                                else
+                                {
+                                    ResourceAction existingAction = resourcePolicy.Actions.First(action => action.Match.Value == resourceAction.Match.Value && action.Match.Id == resourceAction.Match.Id);
+                                    existingAction.RoleGrants.AddRange(resourceAction.RoleGrants.Where(roleGrant => !existingAction.RoleGrants.Any(existingRoleGrant => existingRoleGrant.RoleTypeCode == roleGrant.RoleTypeCode)));
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            return resourcePolicies.Values.ToList();
+        }
+
+        private static List<ResourceAction> GetActionsFromRule(XacmlRule rule, List<RoleGrant> roles)
+        {
+            List<ResourceAction> actions = new List<ResourceAction>();
+            foreach (XacmlAnyOf anyOf in rule.Target.AnyOf)
+            {
+                foreach (XacmlAllOf allOf in anyOf.AllOf)
```

Comment:
Could the ordering here become an issue? Is it possible to create a policy where AllOf comes before AnyOf in the xcaml?

Reply:
No